### ISSUE TITLE
Necessity of C++11 requirements

### ISSUE THREAD

**Issue by FrankHB** [SuperSodaSea/Textcat#2]
Seems not many C++11 features are used overall, even those maybe vital to performance, e.g. move constructors and `noexcept`.

**Comment by SuperSodaSea** [comment]
Yes, I used to prepare to add ```noexcept```, but I was so lazy that I forgot it.
I have heared that move constructors are not necessary for most of the common classes, and the compiler can do it well. Is that true?

**Comment by FrankHB** [comment]
Search `the rules of five`. It won't be defaulted in several cases.
